src/Microsoft.AspNetCore.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -341,7 +358,59 @@ private void BuildFallbackBlock(TagHelperContent builder)
                 .AppendHtml("\",");
 
             AppendFallbackHrefs(builder, fallbackHrefs);
-            builder.AppendHtml("</script>");
+
+            builder.AppendHtml(", \"");
+
+            // Perf: Avoid allocating enumerator
+            for (var i = 0; i < attributes.Count; i++)
+            {
+                var attribute = attributes[i];
+                if (string.Equals(attribute.Name, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                attribute.WriteTo(StringWriter, HtmlEncoder);
+                StringWriter.Write(' ');
+            }
+
+            var stringBuilder = StringWriter.GetStringBuilder();
+            var scriptTags = stringBuilder.ToString();
+            stringBuilder.Clear();
+            var encodedScriptTags = JavaScriptEncoder.Encode(scriptTags);
+            builder.AppendHtml(encodedScriptTags);
+
+            builder.AppendHtml("\");</script>");
+        }
+
+        private bool HasRelStylesheet(TagHelperAttributeList attributes)
+        {
+            TagHelperAttribute relAttribute;
+            if (!attributes.TryGetAttribute(RelAttributeName, out relAttribute) ||
+                relAttribute.Value == null)
+            {
+                return false;
+            }
+
+            var attributeValue = relAttribute.Value;
+            var contentValue = attributeValue as IHtmlContent;
+            var stringValue = attributeValue as string;
+            if (contentValue != null)
+            {
+                contentValue.WriteTo(StringWriter, HtmlEncoder);
+                stringValue = StringWriter.ToString();
+
+                // Reset writer
+                StringWriter.GetStringBuilder().Clear();
+            }
+            else if (stringValue == null)
+            {
+                stringValue = attributeValue.ToString();
```

Comment:
> only called when the value is not a `string` and not an `IHtmlContent`

Got it.

> We considered using the CSS attribute selectors (which expect purely literal content) but that'd be a decently large behavior change.

Agree we considered using attribute selectors but decided to limit the restriction to the fallback handling. The main issue was `<link>` tag helpers lacking `asp-fallback-*` are Just Fine<sup>:tm:</sup> with any random `rel`. That is, the restriction _must_ affect only the `asp-fallback-*` handling.

> I'm just ensuring I can read the `attribute.Value` no matter what it might be. Doesn't hurt to be fool proof here.

As I remember it, we decided _not_ to support `rel='@("stylesheet")'` and so on. Because it does hurt: We have to support these bizarre cases going forward. And we couldn't come up w/ a reasonable scenario in which a user would dynamically choose `stylesheet` or something else.

> It's `object` after all.

My concern is going beyond `string`.